src/devices/cpu/tlcs900/tmp94c241.cpp

```diff
@@ -0,0 +1,1520 @@
+		    for 8-bit timers 0, 1, 2 and 3
+		*/
+		if ( !BIT(m_t8run, i) ) /* Timer isn't running */
+		{
+			m_timer_8[i] = 0;
+			m_timer_change[i] = 0;
+		}
+	}
+}
+
+uint8_t tmp94c241_device::t01mod_r()
+{
+	return m_t01mod;
+}
+
+void tmp94c241_device::t01mod_w(uint8_t data)
+{
+	m_t01mod = data;
+}
+
+uint8_t tmp94c241_device::tffcr_r()
+{
+	return m_tffcr;
+}
+
+enum
+{
+	FF_INVERT,
+	FF_SET,
+	FF_CLEAR,
+	FF_DONTCARE
+};
+
+void tmp94c241_device::change_timer_flipflop(uint8_t flipflop, uint8_t operation)
+{
+	/* First we update the timer flip-flop */
+	bool &ff_state = m_timer_flipflops[flipflop];
+
+	switch(operation)
+	{
+	case FF_INVERT:
+		ff_state = !ff_state;
+		break;
+	case FF_SET:
+		ff_state = true;
+		break;
+	case FF_CLEAR:
+		ff_state = false;
+		break;
+	default:
+		// invalid operation
+		return;
+	}
+
+	/* The value of the flipflop is only exposed to a pin in certain modes of operation
+	   determined by fields of the port function registers.
+
+	   So here we bail out if the flipflop is not routed to its corresponding port bit:
+	*/
+	switch( flipflop )
+		{
+		case 0x1:
+			if (!BIT(m_port_function[PORT_C], 0) || BIT(m_port_control[PORT_C], 0)) return;
+			break;
+		case 0x7:
+			if (!BIT(m_port_function[PORT_C], 0) || !BIT(m_port_control[PORT_C], 0)) return;
+			break;
+		case 0x3:
+			if (!BIT(m_port_function[PORT_C], 1) || BIT(m_port_control[PORT_C], 1)) return;
+			break;
+		case 0xb:
+			if (!BIT(m_port_function[PORT_C], 1) || !BIT(m_port_control[PORT_C], 1)) return;
+			break;
+		case 0x4:
+			if (!BIT(m_port_function[PORT_D], 0)) return;
+			break;
+		case 0x6:
+			if (!BIT(m_port_function[PORT_D], 4)) return;
+			break;
+		case 0x8:
+			if (!BIT(m_port_function[PORT_E], 0)) return;
+			break;
+		case 0xA:
+			if (!BIT(m_port_function[PORT_E], 4)) return;
+			break;
+		default:
+			// invalid flip flop
+			return;
+		}
+
+	/* And here we actually send the value to the corresponding pin */
+	uint8_t new_port_value = 0;
+	switch( flipflop )
+		{
+		case 0x1:
+		case 0x7:
+			new_port_value = m_port_latch[PORT_C] & 0xfe;
+			if (ff_state) new_port_value |= 0x01;
+			port_w<PORT_C>(new_port_value);
+			break;
+		case 0x3:
+		case 0xb:
+			new_port_value = m_port_latch[PORT_C] & 0xfd;
+			if (ff_state) new_port_value |= 0x02;
+			port_w<PORT_C>(new_port_value);
+			break;
+		case 0x4:
+			new_port_value = m_port_latch[PORT_D] & 0xfe;
+			if (ff_state) new_port_value |= 0x01;
+			port_w<PORT_D>(new_port_value);
+			break;
+		case 0x6:
+			new_port_value = m_port_latch[PORT_D] & 0xef;
+			if (ff_state) new_port_value |= 0x10;
+			port_w<PORT_D>(new_port_value);
+			break;
+		case 0x8:
+			new_port_value = m_port_latch[PORT_E] & 0xfe;
+			if (ff_state) new_port_value |= 0x01;
+			port_w<PORT_E>(new_port_value);
+			break;
+		case 0xa:
+			new_port_value = m_port_latch[PORT_E] & 0xef;
+			if (ff_state) new_port_value |= 0x10;
+			port_w<PORT_E>(new_port_value);
+			break;
+		}
```

Comment:
Same here – braces should be at the same level as the `switch` keyword.

Reply:
Thanks!

Is this better now? https://github.com/mamedev/mame/pull/13220/commits/77cb08609ab7bad59a4fbbb8fec2d773a0993395